### PR TITLE
✨ TinaMarkdown - Using it on the Rule template and category pages

### PR DESCRIPTION
**Pull request by Calinator444**
### Description 

This PR uses the TinaMarkdown component on the Category list page and Rule page. I had to format the markdown used by rules so that it would render. The main changes I made to the content are:
- Escaping special characters so that the markdown parser will accept them (e.g.  {, }, ~, and !)
- un-nesting lists
- Removing \<\!\-\-endintro\-\-\> from the markdown body

### Relevant Issue
- Fixed: #1525

### CODE REVIEW

File: schemas/bodySchema.json [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            ]
+            },
+            {
+              "name": "tildes",
+              "label": "Custom Markdown",
+              "match": {
+                "start": "~~",
+                "end": "~~"
+              },
+            "fields": [
```
Comment (isaaclombardssw) [inline schemas/bodySchema.json:90]: All the matches in this file might be able to be removed, since you're now escaping characters.

**Review by isaaclombardssw — APPROVED** [review]
Once you've checked the matches, LGTM.

File: src/templates/rule.js [hunk 2]
```diff
@@ -383,3 +388,7 @@ export const query = graphql`
     }
   }
 `;
+
+const GreyBox = ({ content }) => {
+  return <div id="greybox test">{content}</div>;
+};
```
Comment (bradystroud) [inline src/templates/rule.js:394]: This should live in `components/blocks` 
I know its just a test one but can you move it to the correct place so we can use it as a good example

File: src/templates/rule.js [hunk 3]
```diff
@@ -256,7 +260,8 @@ const Rule = ({ data, location }) => {
                     </div>
                   )}
                 <hr />
-                <div dangerouslySetInnerHTML={{ __html: rule.html }} />
+                <TinaMarkdown content={mdx} components={{ GreyBox }} />
+                {/* <div dangerouslySetInnerHTML={{ __html: rule.html }} /> */}
```
Comment (bradystroud) [inline src/templates/rule.js:264]: 🚮

File: src/templates/rule.js [hunk 4]
```diff
@@ -95,13 +99,13 @@ const Rule = ({ data, location }) => {
...
         </div>
-        <div
+        {/* <div
           style={{
             wordWrap: 'break-word',
             width: 'auto',
           }}
           dangerouslySetInnerHTML={{ __html: props.content }} //Is this a good idea? JS injection ect
-        />
+        /> */}
```
Comment (bradystroud) [inline src/templates/rule.js:108]: Not sure what this is for
If we dont need it then 🚮

File: schemas/bodySchema.json
Comment (bradystroud) [inline schemas/bodySchema.json:1]: should this be in /tina ?

File: package.json [hunk 5]
```diff
@@ -66,7 +66,7 @@
     "gatsby-source-git": "^1.1.0",
     "gatsby-source-local-git": "^1.3.0",
     "gatsby-transformer-json": "^5.13.1",
-    "gatsby-transformer-remark": "3.2.0",
+    "gatsby-transformer-remark": "6.13.1",
```
Comment (bradystroud) [inline package.json:69]: See line 117 - Is this ok to be updated?

File: gatsby-node.js [hunk 6]
```diff
@@ -204,13 +215,16 @@ exports.createPages = async ({ graphql, actions }) => {
       );
     }
 
-    // Create the page for the rule
-    // eslint-disable-next-line no-console
+    // console.log('node.rawMarkdownBody', typeof node.rawMarkdownBody);
+
+    // const md = formatRuleMarkdown(node.rawMarkdownBody);
+
```
Comment (bradystroud) [inline gatsby-node.js:221]: 🚮

File: package.json (hunk 5 above)
Comment (isaaclombardssw) [reply package.json:69]: It seems to work alright on build.

Comment (isaaclombardssw) [reply package.json:69]: Actually let me test it again cause I think it might be pumping out an error atm

**Pull request merged**